packages/core/src/editor/actions.ts

```diff
@@ -200,41 +263,42 @@ export const Actions = (
     setHidden(id: NodeId, bool: boolean) {
       state.nodes[id].data.hidden = bool;
     },
+
     /**
-     * Set custom values to a Node
+     * Update the props of a Node
      * @param id
      * @param cb
      */
-    setCustom<T extends NodeId>(
-      id: T,
-      cb: (data: EditorState["nodes"][T]["data"]["custom"]) => void
-    ) {
-      cb(state.nodes[id].data.custom);
+    setProp(id: NodeId, cb: (props: any) => void) {
+      invariant(state.nodes[id], ERROR_INVALID_NODEID);
+      cb(state.nodes[id].data.props);
     },
-    deserialize(json: string) {
-      const reducedNodes: Record<NodeId, SerializedNodeData> = JSON.parse(json);
-      const rehydratedNodes = Object.keys(reducedNodes).reduce(
-        (accum: Nodes, id) => {
+
+    setState(dehydratedNodes: Record<NodeId, SerializedNodeData>) {
```

Comment:
I think we should fine a better name for this. `setState` gives the impression that we are changing specific state values in the editor, but we're actually replacing the entire state with the serialised data structure. Perhaps we can let this remain `deserialise` and just check if the parameter `dehydratedNodes` is a JSON string before-hand, in which case parse the JSON string and then continue on with the function.